### PR TITLE
Created Joshua Cyberiskvision Solution

### PR DESCRIPTION
**Pull request by cyberiskvision**
First revision of Joshua Cyberiskvision Solution for Microsoft Sentinel developed by Almaviva S.p.A.

For more information visit the following links:
https://www.cyberiskvision.com/
https://www.almaviva.it/en_GB

### CODE REVIEW

**Comment by ritika-msft** [comment]
@anki-narravula please review this playbook. Thanks.

**Comment by anki-narravula** [comment]
Hi @cyberiskvision Thank you for creating solution. Please find my comments below

1. Can you please move data connectors into Data Connectors folder - Solutions/Joshua-Cyberiskvision/Data Connectors/ to avoid confusion

> Solutions/Joshua-Cyberiskvision/Playbooks/Indicator-Processor/Joshua-Import-To-Sentinel.json
> Solutions/Joshua-Cyberiskvision/Playbooks/Indicator-Processor/Joshua-Indicators-Processor-DOMAIN.json
> Solutions/Joshua-Cyberiskvision/Playbooks/Indicator-Processor/Joshua-Indicators-Processor-EMAIL.json

2. Below will be remains in Playbooks folder 

> Solutions/Joshua-Cyberiskvision/Playbooks/Intel-Enrichment/Joshua-Intel-Enrichment-FILE.json
> Solutions/Joshua-Cyberiskvision/Playbooks/Intel-Enrichment/Joshua-Intel-Enrichment-URL.json
> Solutions/Joshua-Cyberiskvision/Playbooks/Intel-Enrichment/Joshua-Intel-Enrichment-IP.json

Below recommendations are for (Enrichment - Playbooks) those are in Playbooks folder. We will update you on Data Connectors soon

3. Unless it is mandatory as per the functionality, we recommend to start the playbook with incident trigger instead of alert trigger. So that you don't require 2 steps ( Alert Trigger --> Get - Incident details). You will get incident date from the Incident trigger itself. Let me know if there is any confusion in this

image

New will be

image


4. As a best practice, we recommend sentinel connection in playbooks uses "ManagedSecurityIdentity".
Please refer Sample [Template](https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json) for sample template. For more details, refer [this](https://techcommunity.microsoft.com/t5/azure-sentinel/what-s-new-managed-identity-for-azure-sentinel-logic-apps/ba-p/2068204). Make sure to do changes at 3 places 

image

image

image

5. Avoid deployment parameters which are really don't require from user perspective. For example Tag and sentinel name can be avoided from deployment parameters

image

6. Rename the deployment file to azuredeploy.json, you can refer to solution for folder structure https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/RiskIQ/Playbooks

7. Sample template available at for reference https://github.com/Azure/Azure-Sentinel/blob/master/Playbooks/.template/incident-trigger/azuredeploy.json

File: Solutions/Joshua-Cyberiskvision/Playbooks/Intel-Enrichment/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Joshua - Cyberiskvision
+
+These playbooks use the Joshua Cyberiskvision threat intelligence to automatically enrich incidents generated by Microsoft Sentinel.
+From the analyst perspective, it is important that the alert contains essential information and therefore the ability to include custom details in the alert will improve the efficiency of investigation.
+Through this integration, the analist can enrich incidents with further information.
```
Comment (NikTripathi) [inline Solutions/Joshua-Cyberiskvision/Playbooks/Intel-Enrichment/README.md:5]: Typo analist -> analyst

**Review by NikTripathi — CHANGES_REQUESTED** [review]

**Comment by cyberiskvision** [comment]
Hi @anki-narravula,
we can't switch trigger from "Alert" to "Incident" because enrichment should be available for manual trigger from alert investigation pane.
All other changes were made successfully.

**Comment by anki-narravula** [comment]
> Hi @anki-narravula, we can't switch trigger from "Alert" to "Incident" because enrichment should be available for manual trigger from alert investigation pane. All other changes were made successfully.

Hi @cyberiskvision  Thanks for making changes. I don't see changes reflected for "Playbooks" (which are in Playbook folder) - like Using system managed account, renaming and removing deployment parameters. Am I missing something ?

**Comment by anki-narravula** [comment]
Hi @v-jayakal  - Now we have data connectors moved to separate folder. Please review data connectors

**Comment by anki-narravula** [comment]
Hi @cyberiskvision  Apologies, there was some confusion for TI category for Data connectors vs Playbooks. We need to go with Playbooks folder only for TI category. Can you please move everything into Playbooks folder and we are good to go.
And as I have recommended earlier, please use managed identity connection for all sentinel related connectors (for authentication with Sentinel) for playbooks "Joshua-Intel-Enrichment-FILE/URI/IP.json".

**Comment by NikTripathi** [comment]
@anki-narravula  Please approve so that we can merge. Thank you.

**Review by anki-narravula — APPROVED** [review]
This looks good. We can merge it.

**Review by v-rucdu — APPROVED** [review]

**Pull request merged**